esp/src/src-react/components/Sasha.tsx

```diff
@@ -1,96 +1,88 @@
 import * as React from "react";
-import { Dropdown, TextField, PrimaryButton } from "@fluentui/react";
+import { Dropdown, IDropdownOption, TextField, PrimaryButton } from "@fluentui/react";
 import nlsHPCC from "src/nlsHPCC";
+import { HolyGrail } from "../layouts/HolyGrail";
+import { SashaService, WsSasha } from "@hpcc-js/comms";
 
 interface SashaProps {}
 
-export const Sasha: React.FunctionComponent<SashaProps> = ({ }) => {
-  const [selectedOption, setSelectedOption] = React.useState("");
-  const [wuid, setWuid] = React.useState("");
-  const [result, setResult] = React.useState("");
+const mySashaService = new SashaService({ baseUrl: "" });
 
-  const handleOptionChange = (event: React.FormEvent<HTMLDivElement>, option: any) => {
-    setSelectedOption(option.key);
+export const Sasha: React.FunctionComponent<SashaProps> = () => {
+  const [selectedOption, setSelectedOption] = React.useState<string>("");
+  const [wuid, setWuid] = React.useState<string>("");
+  const [result, setResult] = React.useState<string>("");
+
+  const handleOptionChange = (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption) => {
+    setSelectedOption(option?.key as string);
   };
 
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
This would need to be wrapped in a `React.useCallback(() => { ... }, [selectedOption, wuid]);`